src/components/VmDetails/cards/OverviewCard/index.js

```diff
@@ -138,6 +169,11 @@ class OverviewCard extends React.Component {
       vmUpdates['description'] = stateVm.get('description')
     }
 
+    if (this.trackUpdates['name'] && updateCloudInit && this.isCloudInitHostnameUpdate()) {
+      vmUpdates['cloudInit'] = stateVm.get('cloudInit').toJS()
```

Comment:
Is the full structure necessary to partially update the cloudInit hostname?  Would it work with something like:
````js
vmUpdates['cloudInit'] = { hostName: stateVm.get('name') }
````

Reply:
It must have full structure, because it will update `cloudInit` only if cloud init is `enabled`, thus it better to leave it as it is.

Reply:
It is ok to copy it forward the entire structure and just update the bit that needs it.  A bit of a deviation from the "update exactly only what is necessary to update", but that's ok.  No need to change.